src/overlays/actors/ovl_Obj_Chan/z_obj_chan.c

```diff
@@ -31,49 +39,404 @@ const ActorInit Obj_Chan_InitVars = {
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Chan/ObjChan_Destroy.s")
+    dl_opa = Gfx_CallSetupDL(POLY_OPA_DISP, 0x19U);
+    gSPMatrix(dl_opa, Matrix_NewMtx(globalCtx->state.gfxCtx), G_MTX_LOAD);
+    gSPDisplayList(dl_opa + 1, object_obj_chan_DL_000AF0);
+    POLY_OPA_DISP = dl_opa + 2;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Chan/func_80BB9A1C.s")
+    dl_xlu = func_8012C2B4(POLY_XLU_DISP);
+    gSPMatrix(dl_xlu, Matrix_NewMtx(globalCtx->state.gfxCtx), G_MTX_LOAD);
+    gSPDisplayList(dl_xlu + 1, object_obj_chan_DL_000A10);
+    POLY_XLU_DISP = dl_xlu + 2;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Chan/func_80BB9B40.s")
+    CLOSE_DISPS(globalCtx->state.gfxCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Chan/func_80BB9C08.s")
+    Matrix_StatePush();
+    if (this->stateFlags & OBJCHAN_STATE_ON_FIRE) {
+        func_80BBA930(this, globalCtx);
+    }
+    Matrix_StatePop();
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Chan/func_80BB9F24.s")
+void ObjChan_DrawPot(Actor* thisx, GlobalContext* globalCtx) {
+    ObjChan* this = THIS;
+    s32 pad;
+    Gfx* dl;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Chan/func_80BBA2FC.s")
+    OPEN_DISPS(globalCtx->state.gfxCtx);
+    dl = Gfx_CallSetupDL(POLY_OPA_DISP, 0x19U);
+    gSPMatrix(dl, Matrix_NewMtx(globalCtx->state.gfxCtx), G_MTX_LOAD);
+    gSPDisplayList(dl + 1, object_obj_chan_DL_002358);
+    POLY_OPA_DISP = dl + 2;
+    CLOSE_DISPS(globalCtx->state.gfxCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Chan/func_80BBA314.s")
+    if (this->stateFlags & OBJCHAN_STATE_ON_FIRE) {
+        func_80BBA930(this, globalCtx);
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Chan/func_80BBA488.s")
+void func_80BBA930(ObjChan* this, GlobalContext* globalCtx) {
+    u32 sp4C;
+    Gfx* dl;
+    OPEN_DISPS(globalCtx->state.gfxCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Chan/ObjChan_Update.s")
+    sp4C = globalCtx->gameplayFrames;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Chan/ObjChan_Draw.s")
+    Matrix_RotateY(func_800DFCDC(GET_ACTIVE_CAM(globalCtx)) - this->actor.shape.rot.y - this->rotation + 0x8000, 1);
+    Matrix_Scale(sObjChanFlameSize[OBJCHAN_SUBTYPE(this)].x * this->flameSize,
+                 sObjChanFlameSize[OBJCHAN_SUBTYPE(this)].y * this->flameSize, 1.0f, 1);
+    Matrix_InsertTranslation(0.0f, sObjChanFlameYOffset[OBJCHAN_SUBTYPE(this)], 0.0f, 1);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Obj_Chan/func_80BBA894.s")
+    dl = func_8012C2B4(POLY_XLU_DISP);
+    gSPMatrix(dl, Matrix_NewMtx(globalCtx->state.gfxCtx), G_MTX_LOAD);
+    gMoveWd(dl + 1, 6, 32,
+            Gfx_TwoTexScroll(globalCtx->state.gfxCtx, 0, 0U, 0U, 0x20, 0x40, 1, 0U, -sp4C * 20, 0x20, 0x80));
+    gDPSetPrimColor(dl + 2, 0x80, 0x80, 0xFF, 0xFF, 0x00, 0xFF);
+    gDPSetEnvColor(dl + 3, 0xFF, 0x00, 0x00, 0x00);
```

Comment:
These are RGB values and can be changed to decimal 
```suggestion
    gDPSetPrimColor(dl + 2, 128, 128, 255, 255, 0, 255);
    gDPSetEnvColor(dl + 3, 255, 0, 0, 0);
```
These are RBG / color values and can use decimal values instead of hex